internal/driven/encryption/jwt/encryption.go

```diff
@@ -1,40 +1,55 @@
+// Package encryption provides functionality for JWT token generation and validation,
+// as well as password hashing and checking hashed passwords.
 package encryption
 
 import (
 	"github.com/dgrijalva/jwt-go"
+	"github.com/isdzulqor/donation-hub/internal/core/entity"
 	"golang.org/x/crypto/bcrypt"
 	"time"
 )
 
+// JWTService is an interface that defines methods for generating and validating JWT tokens.
 type JWTService interface {
-	GenerateToken(userID int64, role string) (string, error)
+	// GenerateToken generates a JWT token for a given user.
+	GenerateToken(user entity.User) (string, error)
+	// ValidateToken validates a given JWT token and returns the parsed token if it's valid.
 	ValidateToken(token string) (*jwt.Token, error)
 }
 
+// jwtService is a struct that implements the JWTService interface.
 type jwtService struct {
 	secretKey string
 	issuer    string
 }
 
+// NewJWTService creates a new JWTService with the given secret key and issuer.
 func NewJWTService(secretKey string, issuer string) JWTService {
 	return &jwtService{
 		secretKey: secretKey,
 		issuer:    issuer,
 	}
 }
 
+// CustomClaims is a struct that defines the custom claims for the JWT token.
 type CustomClaims struct {
-	UserID int64  `json:"user_id"`
-	Role   string `json:"role"`
+	ID       int64  `json:"id"`
+	Username string `json:"username"`
+	Email    string `json:"email"`
 	jwt.StandardClaims
 }
 
-func (s *jwtService) GenerateToken(userID int64, role string) (string, error) {
+// GenerateToken generates a JWT token for a given user.
+// The token includes custom claims (user ID, username, and email) and standard claims (expiry time and issuer).
+func (s *jwtService) GenerateToken(user entity.User) (string, error) {
+
+	// one year expiry time token
 	claims := &CustomClaims{
-		UserID: userID,
-		Role:   role,
+		ID:       user.ID,
+		Username: user.Username,
+		Email:    user.Email,
 		StandardClaims: jwt.StandardClaims{
-			ExpiresAt: time.Now().Add(time.Hour * 72).Unix(),
+			ExpiresAt: time.Now().Add(time.Hour * 24 * 365).Unix(),
```

Comment:
Put the token duration config in ENVAR. So, it's easily to adjust in the future.